src/inference/src/os/mac/mac_system_conf.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <sys/sysctl.h>
+
+#include <memory>
+#include <vector>
+
+#include "dev/threading/parallel_custom_arena.hpp"
+#include "openvino/runtime/system_conf.hpp"
+#include "streams_executor.hpp"
+
+namespace ov {
+
+CPU::CPU() {
+    _num_threads = parallel_get_max_threads();
+    parse_processor_info_macos(
+                             _processors,
+                             _numa_nodes,
+                             _cores,
+                             _proc_type_table);
+}
+
+int parse_processor_info_macos(int& _processors,
+                               int& _numa_nodes,
+                               int& _cores,
+                               std::vector<std::vector<int>>& _proc_type_table) {
+    uint64_t output = 0;
+    size_t size = sizeof(output);
+
+    _processors = 0;
+    _numa_nodes = 0;
+    _cores = 0;
+
+    if (sysctlbyname("hw.ncpu", &output, &size, NULL, 0) < 0) {
+        return -1;
+    } else {
+        _processors = output;
+    }
+
+    if (sysctlbyname("hw.physicalcpu", &output, &size, NULL, 0) < 0) {
+        _processors = 0;
+        return -1;
+    } else {
+        _cores = output;
```

Comment:
`_cores `type is `int`, while `output `is `uint64`, it will report error if warning_as_error build option is enabled.
We need add explicit type conversion to avoid it.
Please clean all such issues in this PR.

Reply:
I have added explicit type conversion.